src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -314,6 +330,37 @@ public WorkspaceAnalyzerConfigOptions(AnalyzerConfigOptionsResult analyzerConfig
             }
         }
 
+        private sealed class WorkspaceSyntaxTreeOptionsProvider : SyntaxTreeOptionsProvider
+        {
+            private readonly ValueSource<AnalyzerConfigSet> _lazyAnalyzerConfigSet;
+
+            public WorkspaceSyntaxTreeOptionsProvider(ValueSource<AnalyzerConfigSet> lazyAnalyzerConfigSet)
+                => _lazyAnalyzerConfigSet = lazyAnalyzerConfigSet;
+
+            public override bool? IsGenerated(SyntaxTree tree)
+            {
+                var options = _lazyAnalyzerConfigSet
+                    .GetValue(CancellationToken.None).GetOptionsForSourcePath(tree.FilePath);
```

Comment:
Any reason we shouldn't thread cancellation through the SyntaxTreeOptionsProvider?

Reply:
Oversight. I'll fix.

Reply:
Sorry, didn't fully understand the implications here -- now that I think about it, I think it's better not to do so, and the reason is because once you've supplied the provider to the compiler, there are a variety of operations which are simply uncancellable that will request the provider. I think it's best to leave it out because it incorrectly implies that the cancellation token is respected in the compiler when it won't be.

Reply:
> there are a variety of operations which are simply uncancellable that will request the provider

What operations would be uncancellable?

> I think it's best to leave it out because it incorrectly implies that the cancellation token is respected in the compiler when it won't be.

Cancellation is _always_ advisory so the compiler is allowed to do so. My concern is if/when this becomes a perf issue then we will have to do more API evilness to deal with that.

Reply:
I'll address the cancellation concern in a separate pull request.